### PR TITLE
Parse container ID and host URL from driver log rather than YARN UI

### PR DESCRIPTION
**Pull request by t-rufang**
To address #2866

### CODE REVIEW

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchJob.java [hunk 1]
```diff
@@ -1040,6 +919,28 @@ public boolean isActive() throws IOException {
                 .map(yarnApp -> yarnApp.getLogAggregationStatus().toUpperCase());
     }
 
+    public Observable<Integer> getYarnContainerLogUrlPort() {
+        final int DEFAULT_YARN_CONTAINER_LOG_URL_PORT = 30060;
+        return getSparkJobApplicationIdObservable()
```
Comment (wezhang) [inline Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchJob.java]: Could you implement this as a lazy evaluation?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobDebuggerRunner.java [hunk 2]
```diff
@@ -228,6 +230,26 @@ protected void execute(ExecutionEnvironment environment, Callback callback, RunP
...
+                            // Get YARN container log URL port
+                            int containerLogUrlPort =
+                                    ((SparkBatchRemoteDebugJob) driverDebugProcess.getSparkJob())
+                                            .getYarnContainerLogUrlPort()
+                                            .toBlocking()
+                                            .single();
+                            // Parse container ID and host URL from driver console view
+                            jobOutputView.getSecondaryConsoleView().addMessageFilter((line, entireLength) -> {
+                                Matcher matcher = Pattern.compile(
+                                        "Launching container (\\w+)( for|) on host ([a-zA-Z_0-9-.]+)",
```
Comment (wezhang) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobDebuggerRunner.java]: How about this:
```suggestion
                                        "Launching container (\\w+) .* on host ([a-zA-Z_0-9-.]+)",
```

File: Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchJob.java (hunk 1 above)
Comment (t-rufang) [reply Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/common/SparkBatchJob.java]: `getYarnContainerLogUrlPort` is only called once so lazy evaluation is not necessary.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobDebuggerRunner.java (hunk 2 above)
Comment (t-rufang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobDebuggerRunner.java]: Changed in the new commit.

**Pull request merged**